plugins/protocols/wayfire-shell.cpp

```diff
@@ -448,6 +448,11 @@ class wayfire_shell_protocol_impl : public wf::plugin_interface_t
 
     void fini() override
     {
+        if (wf::get_core().get_current_state() == wf::compositor_state_t::SHUTDOWN)
```

Comment:
Why is this necessary? Is this something all protocol plugins need to do, for example gtk-shell as well?